### PR TITLE
Reporting mode api

### PR DESCRIPTION
**Pull request by mDuo13**
Changes to the API reference docs for Reporting Mode ( https://github.com/ripple/rippled/pull/3609 )

### CODE REVIEW

File: content/references/rippled-api/api-conventions/basic-data-types.md [hunk 1]
```diff
@@ -68,17 +68,30 @@ Some types of hash appear in API requests and responses. Others are only calcula
...
+If you do not specify a ledger, the server decides which ledger to use to serve the request. By default, the server chooses the `current` (in-progress) ledger. In [Reporting Mode](rippled-server-modes.html#reporting-mode), the server uses the most recent validated ledger instead. Do not provide more than one field specifying ledgers, the server may choose any of them to . The `ledger_index` field is ignored unless neither of the other two are present.
+
+**Note:** Do not rely on the default behavior for specifying a ledger; it is subject to change. Always specify a ledger version in the request if you can.
 
-**Note:** Do not rely on this default behavior for specifying a ledger; it is subject to change. Always specify a ledger version in the request if you can.
+If you make a request to a Reporting Mode server for a ledger that has not been validated, the server forwards the request to a network mode server, because Reporting Mode does not record ledger data until it has been validated.
```
Comment (cjcobb23) [inline content/references/rippled-api/api-conventions/basic-data-types.md]: This is not 100% true. If you specify `current` or `closed`, the reporting server will forward to a p2p server. But if you just specify a ledger index or hash that the reporting server does not have, it just returns `lgrNotFound`

File: content/references/rippled-api/public-rippled-methods/server-info-methods/fee.md [hunk 2]
```diff
@@ -157,7 +157,8 @@ The response follows the [standard format][], with a successful result containin
 
 ## Possible Errors
 
-* Any of the [universal error types][].
+- Any of the [universal error types][].
+- `reportingUnsupported` - ([Reporting Mode][] servers only) This method is not available in Reporting Mode.
```
Comment (cjcobb23) [inline content/references/rippled-api/public-rippled-methods/server-info-methods/fee.md]: `fee` is available in reporting mode. The reporting server will just forward the request to the p2p node. If that fails, the error will be `failedToForward`.

File: content/references/rippled-api/api-conventions/basic-data-types.md (hunk 1 above)
Comment (cjcobb23) [reply content/references/rippled-api/api-conventions/basic-data-types.md]: I think also "network mode server" should be changed to "P2P mode server", to be consistent with the other doc talking about the modes.

**Review by cjcobb23 — CHANGES_REQUESTED** [review]
Looks good. Just a few small things that I think need to be changed.

File: content/references/rippled-api/api-conventions/basic-data-types.md [hunk 3]
```diff
@@ -68,17 +68,30 @@ Some types of hash appear in API requests and responses. Others are only calcula
...
+            "ledger_index": "validated"
+
     * `closed` for the most recent ledger that has been closed for modifications and proposed for validation
+
     * `current` for the server's current working version of the ledger.
 
 There is also a deprecated `ledger` parameter which accepts any of the above three formats. *Do not* use this parameter; it may be removed without further notice.
 
-If you do not specify a ledger, the `current` (in-progress) ledger is chosen by default. If you provide more than one field specifying ledgers, the deprecated `ledger` field is used first if it exists, falling back to `ledger_hash`. The `ledger_index` field is ignored unless neither of the other two are present.
+If you do not specify a ledger, the server decides which ledger to use to serve the request. By default, the server chooses the `current` (in-progress) ledger. In [Reporting Mode](rippled-server-modes.html#reporting-mode), the server uses the most recent validated ledger instead. Do not provide more than one field specifying ledgers, the server may choose any of them to . The `ledger_index` field is ignored unless neither of the other two are present.
```
Comment (ddibble-rip) [inline content/references/rippled-api/api-conventions/basic-data-types.md]: Incomplete sentence here: 
Do not provide more than one field specifying ledgers, the server may choose any of them to <something_should_go_here>.

File: content/references/rippled-api/api-conventions/error-formatting.md [hunk 4]
```diff
@@ -101,15 +101,16 @@ For other errors that returned with HTTP status code 200 OK, the responses are f
...
-* `noCurrent` - The server does not know what the current ledger is, due to high load, network problems, validator failures, incorrect configuration, or some other problem.
-* `noNetwork` - The server is having trouble connecting to the rest of the XRP Ledger peer-to-peer network (and is not running in stand-alone mode).
-* `tooBusy` - The server is under too much load to do this command right now. Generally not returned if you are connected as an admin.
-* `unknownCmd` - The request does not contain a [command](rippled-api.html) that the `rippled` server recognizes.
-* `wsTextRequired` - (WebSocket only) The request's [opcode](https://tools.ietf.org/html/rfc6455#section-5.2) is not text. <!-- SPELLING_IGNORE: opcode -->
+- `jsonInvalid` - (WebSocket only) The request is not a proper JSON object.
+    - JSON-RPC returns a 400 Bad Request HTTP error in this case instead.
+- `missingCommand` - (WebSocket only) The request did not specify a `command` field.
+    - JSON-RPC returns a 400 Bad Request HTTP error in this case instead.
+- `noClosed` - The server does not have a closed ledger, typically because it has not finished starting up.
```
Comment (ddibble-rip) [inline content/references/rippled-api/api-conventions/error-formatting.md:111]: Not sure if the above JSON-RPC bullet entries render correctly due to indentation.

Comment (mDuo13) [reply content/references/rippled-api/api-conventions/error-formatting.md:111]: Confirmed it renders as intended:

**Review by cjcobb23 — APPROVED** [review]
Looks good.

**Pull request merged**